spec/models/links_udpate_spec.rb

```diff
@@ -53,28 +53,43 @@
   end
 
   describe '#mark_as_errored' do
-    let(:tag_mapping) { create(:tag_mapping) }
+    let!(:tag_mapping) { create(:tag_mapping) }
 
     before do
-      links_update.tag_mappings = TagMapping.all
+      publishing_api_has_lookups('/my-base-path' => 'my-content-id')
+
+      publishing_api_has_linkables(
+        [
+          { 'title' => 'Early Years', content_id: 'my-content-id' },
+        ],
+        document_type: 'taxon'
+      )
     end
 
-    it 'marks a number of tag mappings as errored' do
-      expectation = lambda do
-        links_update.mark_as_errored
-        tag_mapping.reload
-      end
+    it 'adds error message for invalid link type' do
+      links_update = build(:links_update, links: { 'invalid' => ['my-content-id'] })
+      links_update.tag_mappings = TagMapping.all
 
-      expect { expectation.call }.to change { tag_mapping.state }.to('errored')
+      links_update.valid?
+      links_update.mark_as_errored
+      tag_mapping.reload
+
+      expect(tag_mapping.state).to eql('errored')
+      expect(tag_mapping.message).to_not match(/^Link types/)
```

Comment:
Right, I follow your reasoning. I think the underlying issue here is that the test for `#mark_as_errored` is difficult to follow and a little confusing in its current form. For example - both of the test cases trigger both of the validation errors, but we only match on one message at a time and there's test setup present that seems to have no implication for the actual end result.

I've had a quick chat with @carvil and have taken a slightly different approach. Given we already have tests for the validators on LinksUpdate elsewhere, the spec for `#mark_as_errored` can concentrate on the error message concatenation and state change behaviour. Pushing a commit now - WDYT?